### PR TITLE
initial support for pytorch

### PR DESCRIPTION
**Pull request by lgarithm**
Enable primitive support for PyTorch:
- only float32 support
- must install kungfu-tensorflow then install kungfu-pytorch. This we need the kungfu-tensorflow to build the gaoling part of KungFu.

### CODE REVIEW

**Comment by luomai** [comment]
> $ horovodrun -np 4 python3 benchmark_horovod_torch.py 
> 2020-08-01 15:24:16.632634: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
> [1,0]<stdout>:Model: resnet50
> [1,0]<stdout>:Batch size: 32
> [1,0]<stdout>:Number of GPUs: 4
> [1,0]<stdout>:Running warmup...
> [1,0]<stdout>:Running benchmark...
> [1,0]<stdout>:Iter #0: 111.8 img/sec per GPU
> [1,0]<stdout>:Iter #1: 97.2 img/sec per GPU
> [1,0]<stdout>:Iter #2: 97.9 img/sec per GPU
> [1,0]<stdout>:Iter #3: 98.2 img/sec per GPU
> [1,0]<stdout>:Iter #4: 99.0 img/sec per GPU
> [1,0]<stdout>:Iter #5: 103.6 img/sec per GPU
> [1,0]<stdout>:Iter #6: 100.5 img/sec per GPU
> [1,0]<stdout>:Iter #7: 94.9 img/sec per GPU
> [1,0]<stdout>:Iter #8: 94.9 img/sec per GPU
> [1,0]<stdout>:Iter #9: 93.1 img/sec per GPU
> [1,0]<stdout>:Img/sec per GPU: 99.1 +-10.0
> [1,0]<stdout>:Total img/sec on 4 GPU(s): 396.5 +-40.0

**Comment by luomai** [comment]
```
$ kungfu-run -np 4 python3 benchmark_kungfu_torch.py 
[arg] [0]=kungfu-run
[arg] [1]=-np
[arg] [2]=4
[arg] [3]=python3
[arg] [4]=benchmark_kungfu_torch.py
[nic] [0] lo :: 127.0.0.1/8, ::1/128
[nic] [1] eth0 :: 146.179.131.193/25, fe80::ec4:7aff:fe80:9d4/64
[nic] [2] eth1 :: fe80::ec4:7aff:fe80:9d5/64
[nic] [3] eth10 :: fe80::fe15:b4ff:fe43:c870/64
[nic] [4] eth3 :: 
[nic] [5] docker0 :: 172.17.0.1/16, fe80::42:3ff:fefc:a46/64
[I] will parallel run 4 instances of python3 with ["benchmark_kungfu_torch.py"]
[127.0.0.1.10003::stderr] [W] TODO supoort BINARY_TREE_STAR for genCrossStrategyList. using RING
[127.0.0.1.10002::stderr] [W] TODO supoort BINARY_TREE_STAR for genCrossStrategyList. using RING
[127.0.0.1.10000::stderr] [W] TODO supoort BINARY_TREE_STAR for genCrossStrategyList. using RING
[127.0.0.1.10001::stderr] [W] TODO supoort BINARY_TREE_STAR for genCrossStrategyList. using RING
[127.0.0.1.10000::stdout] Model: resnet50
[127.0.0.1.10000::stdout] Batch size: 32
[127.0.0.1.10000::stdout] Number of GPUs: 4
[127.0.0.1.10000::stdout] Running warmup...
[127.0.0.1.10000::stdout] Running benchmark...
[127.0.0.1.10000::stdout] Iter #0: 88.2 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #1: 88.8 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #2: 88.2 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #3: 89.1 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #4: 88.2 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #5: 89.1 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #6: 89.3 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #7: 88.2 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #8: 87.9 img/sec per GPU
[127.0.0.1.10000::stdout] Iter #9: 89.8 img/sec per GPU
[127.0.0.1.10000::stdout] Img/sec per GPU: 88.7 +-1.2
[127.0.0.1.10000::stdout] Total img/sec on 4 GPU(s): 354.7 +-4.7
[I] all 4/4 local peers finished, took 46.555972812s
```
